src/hotspot/share/gc/serial/serialFullGC.cpp

```diff
@@ -705,6 +705,16 @@ void SerialFullGC::invoke_at_safepoint(bool clear_all_softrefs) {
 
   allocate_stacks();
 
+  // Usually, all classunloading works occurs at the end of phase 1, but Serial
+  // full-gc accesses dead-objs' klass to find out the start of next live-obj
+  // during phase 2. This requires klasses of dead-objs to be kept loaded.
+  // Therefore, we declare classunloading context in the same level as various
+  // phases, and purge dead classes (invoking ClassLoaderDataGraph::purge)
+  // after all phases of full-gc.
```

Comment:
```suggestion
  // phases, and purge dead classes (invoking ClassLoaderDataGraph::purge())
  // after the other phases of full-gc.
```

Reply:
Not sure why "full gc" has a hyphen in the middle but I guess that's done elsewhere too.